src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -507,5 +520,26 @@ public FormValidation doCheckHealthScaleFactor(@QueryParameter double value) {
                     (int) (100.0 - Math.max(0.0, Math.min(100.0, 5 * value)))
             ));
         }
+
+        public ListBoxModel doFillStdioRetentionItems(@QueryParameter("stdioRetention") String value) {
```

Comment:
This is not exposed from the **Pipeline Syntax** page, which just shows an empty dropdown (with no help); I think you forgot to delegate to this from `JUnitResultsStep.DescriptorImpl`.